packages/shared/src/store/useChannelPosts.ts

```diff
@@ -177,6 +190,27 @@ export const useChannelPosts = (options: UseChanelPostsParams) => {
     return newestQueryPostId ? [...newerPosts, ...dedupedQueryPosts] : newPosts;
   }, [query.data, query.hasPreviousPage, newPosts]);
 
+  const rawPosts = useMemo(() => {
```

Comment:
Feels like this could be combined with the above dedupe logic? But maybe I'm missing something.

Reply:
I can combine the two `useMemo`s, which might save cognitive load

but if you're saying that we could merge `newPosts` with `unconfirmedPosts` and use the same dedupe logic, then I think this early exit breaks it: https://github.com/tloncorp/tlon-apps/blob/26429d11f34aa476bb0bcbe2c073bc9e0da1a457/packages/shared/src/store/useChannelPosts.ts#L172-L174
My reading of that code is that we only want to show posts coming in through subscription once the channel is completely backfilled; which conflicts with unconfirmed behavior, which we want to show as soon as possible.

will add a comment about this – I don't see a clean way to unify these, lmk if you do

Reply:
https://github.com/tloncorp/tlon-apps/blob/859bd4f92bdcd8d78859455dcdc76b87a9d22fb4/packages/shared/src/store/useChannelPosts.ts#L200-L208